From 548a4250358278bc1b40851d2e452b9e09d3d1f6 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 20 May 2021 08:35:22 +0300
Subject: [PATCH] [SW-42558] gaudi: remove iatu/dbi access toggling

In secure device mode, the iATU access is no more needed as FW is
configuring the iATU by itself and thus also the access to its
configuration is restricted.
Remove the iATU access configuration logic.

Change-Id: I7dc195c7db0ef850784c4c1e3e21b3bc3d493926
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 3e1367b..e04bc54 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -454,18 +454,6 @@
 	return mmPSOC_EFUSE_READ;
 }
 
-
-void preboot_conf_iatu_access(bool enable)
-{
-	u32 reg;
-
-	if (!preboot_security_is_enabled())
-		return;
-
-	reg = enable ? 0x00300000 : 0;
-	regmap_writel(reg, mmPCIE_AUX_DBI);
-}
-
 #define GAUDI_IRQ_SIZE	512
 void preboot_ack_razwi_irq(void)
 {
@@ -484,7 +472,6 @@
 		return;
 
 	preboot_security_dbi_lock();
-	preboot_conf_iatu_access(false);
 	preboot_ack_razwi_irq();
 	iatu_disabled = true;
 }
diff --git a/include/preboot.h b/include/preboot.h
index 0bd441e..424ca0c 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -166,7 +166,6 @@
 void preboot_dev_enable_caps(void);
 void preboot_set_enabled_cap(enum hl_fw_en_caps en_cap);
 void preboot_finalize_enabled_caps(void);
-void preboot_conf_iatu_access(bool enable);
 void preboot_disable_iatu_access(void);
 void preboot_ack_razwi_irq(void);
 const char *preboot_get_version(enum preboot_vers ver);
diff --git a/preboot/preboot.c b/preboot/preboot.c
index e0f53c1..1bce401 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -336,7 +336,6 @@
 
 __weak void hl_dev_pll_init(void) {}
 __weak void hl_clock_gating_init(void) {}
-__weak void preboot_conf_iatu_access(bool enable) {}
 __weak void preboot_process_last_rma_info(void) {}
 
 void preboot_main(void)
@@ -368,9 +367,6 @@
 
 	/* Get the VendorID before we lock the access to configuration space */
 	preboot_retrieve_device_id_from_pcie();
-
-	/* Enable access to iATU before locking PCIE_AUX */
-	preboot_conf_iatu_access(true);
 
 	preboot_security_init();
 	preboot_security_dynamic_init();
